src/JustSaying/Messaging/Channels/Context/QueueMessageContext.cs

```diff
@@ -9,17 +9,17 @@ namespace JustSaying.Messaging.Channels.Context
     /// <inheritdoc />
     public class QueueMessageContext : IQueueMessageContext
     {
-        private ISqsQueue _sqsQueue;
+        private readonly SqsQueueReader _queueReader;
 
         /// <summary>
         /// A handle for a given message to be deleted from the queue it was read from.
         /// </summary>
         /// <param name="message">The <see cref="Amazon.SQS.Model.Message"/> to be handled.</param>
         /// <param name="sqsQueue">The <see cref="ISqsQueue"/> the message was read from.</param>
-        public QueueMessageContext(Message message, ISqsQueue sqsQueue)
+        internal QueueMessageContext(Message message, SqsQueueReader queueReader)
```

Comment:
What do we gain from preventing users of this library from instantiating this type? Or why not make the whole type `internal`?

Reply:
SqsQueueReader can be internal, and we don't want to expose the methods on there

Reply:
Maybe we just make the whole type internal then?

Reply:
It's interface `IQueueMessageContext` is used by [IMessageDispatcher](https://github.com/justeat/JustSaying/blob/5efa352726e7d41c4ba94a5c4a84fc0837cb6ff2/src/JustSaying/AwsTools/MessageHandling/Dispatch/IMessageDispatcher.cs#L18), which is public because `MessageDispatcher` is an existing class that was public before.